src/controller/CHIPDeviceController.cpp

```diff
@@ -2018,9 +2070,15 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         // Read the requested minimum connection times from all network commissioning clusters
         readPaths[7] = app::AttributePathParams(app::Clusters::NetworkCommissioning::Id,
                                                 app::Clusters::NetworkCommissioning::Attributes::ConnectMaxTimeSeconds::Id);
+        // Read the current fabrics
+        if (params.GetCheckForMatchingFabric().ValueOr(false))
+        {
+            readPaths[8] = app::AttributePathParams(OperationalCredentials::Id, OperationalCredentials::Attributes::Fabrics::Id);
+        }
 
         readParams.mpAttributePathParamsList    = readPaths;
-        readParams.mAttributePathParamsListSize = 8;
+        readParams.mAttributePathParamsListSize = 8 + (params.GetCheckForMatchingFabric().ValueOr(false) ? 1 : 0);
```

Comment:
This might be somewhat clearer/safer if we set `readParams.mAttributePathParamsListSize = 8` before we  check `params.GetCheckForMatchingFabric().ValueOr(false)` and then in that block when we seat `readPaths[8]` we also set `readParams.mAttributePathParamsListSize = 9`.  That way we don't have to match up the conditions across different lines, etc.